### PR TITLE
[Ide] Add "Set startup projects" command

### PR DESCRIPTION
**Pull request by rodrmoya**
Fixes https://devdiv.visualstudio.com/DevDiv/_workitems/edit/669199

### CODE REVIEW

**Comment by rodrmoya** [comment]
@sayedihashimi feel free to have a try at this and see if it fully matches the spec, please! There's one issue right now, when opening the preferences dialog, the run configurations panel is not selected as it should, working on it. But apart from that, I think I haven't missed anything

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/ProjectPadContextMenu.addin.xml [hunk 1]
```diff
@@ -152,6 +152,9 @@
 	<Condition id="ItemType" value="IBuildTarget">
 		<CommandItem id = "MonoDevelop.Ide.Commands.ProjectCommands.Options" />
 	</Condition>
+	<Condition id="ItemType" value="Solution|Workspace">
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Gui.Pads.ProjectPad/ProjectPadContextMenu.addin.xml]: This command only applies to solutions, not workspaces.

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs [hunk 2]
```diff
@@ -131,6 +132,37 @@ protected override void Run ()
 		}
 	}
 
+	internal class SetStartupProjectsHandler : CommandHandler
+	{
+		protected override void Update (CommandInfo info)
+		{
+			info.Enabled = IdeApp.ProjectOperations.CurrentSelectedSolution?.GetAllProjects ().Count () > 0;
```
Comment (slluis) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs]:
```suggestion
			info.Enabled = IdeApp.ProjectOperations.CurrentSelectedSolution?.GetAllProjects ().Any ();
```

Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs]: right, this was wrong, it was supposed to be `> 1` so fixed now

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs [hunk 3]
```diff
@@ -131,6 +132,37 @@ protected override void Run ()
 		}
 	}
 
+	internal class SetStartupProjectsHandler : CommandHandler
+	{
+		protected override void Update (CommandInfo info)
+		{
+			info.Enabled = IdeApp.ProjectOperations.CurrentSelectedSolution?.GetAllProjects ().Count () > 1;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs]:
```suggestion
			info.Enabled = IdeApp.ProjectOperations.CurrentSelectedSolution?.GetAllProjects ().Skip (1).Any ();
```

File: main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs [hunk 4]
```diff
@@ -131,6 +132,38 @@ protected override void Run ()
...
+			info.Enabled = IdeApp.ProjectOperations.CurrentSelectedSolution?.GetAllProjects ()?.Skip (1)?.Any () ?? false;
+		}
+
+		protected override void Run ()
+		{
+			var sol = IdeApp.ProjectOperations.CurrentSelectedSolution;
+			if (sol != null) {
+				MultiItemSolutionRunConfiguration config = null;
+				if (!sol.MultiStartupRunConfigurations.Any ()) {
+					using (var dlg = new NewSolutionRunConfigurationDialog ()) {
```
[the remark this answers is not in the document]
Comment (rodrmoya) [reply main/src/core/MonoDevelop.Ide/MonoDevelop.Ide.Commands/ProjectCommands.cs]: Had to patch Xwt (https://github.com/mono/xwt/pull/910) for A11Y issues, but now everything seems to work running it natively, so all ready here

**Pull request merged**